src/lib/checkbox/checkbox-required-validator.ts

```diff
@@ -0,0 +1,48 @@
+import {
+  Input,
+  Directive,
+  forwardRef,
+  Provider,
+} from '@angular/core';
+import {
+  AbstractControl,
+  ValidationErrors,
+  Validator,
+  Validators,
+  NG_VALIDATORS,
+} from '@angular/forms';
+
+export const MD_CHECKBOX_REQUIRED_VALIDATOR: Provider = {
+  provide: NG_VALIDATORS,
+  useExisting: forwardRef(() => MdCheckboxRequiredValidator),
+  multi: true
+  };
+
+@Directive({
+  selector:
+  'md-checkbox[required][formControlName],' +
+  'md-checkbox[required][formControl],md-checkbox[required][ngModel]',
+  providers: [MD_CHECKBOX_REQUIRED_VALIDATOR],
+  host: {'[attr.required]': 'required ? "" : null'}
+})
+export class MdCheckboxRequiredValidator implements Validator {
```

Comment:
Considered but AOT doesn't support component inheritance. 
This is the error I got when "extends CheckboxRequiredValidator`: https://travis-ci.org/angular/material2/jobs/228945376